Build failure with text-2.1.2

### Summary

When built with `text-2.1.2` (uploaded 2024-10-26T13:22:01Z) in the package set, the following errors occur:

```
src/PlutusLedgerApi/Common/Eval.hs:117:34: error: [GHC-87543]
    Ambiguous occurrence ‘show’.
    It could refer to
       either ‘PlutusCore.show’,
              imported from ‘PlutusCore’ at src/PlutusLedgerApi/Common/Eval.hs:23:1-17
              (and originally defined in ‘GHC.Show’),
           or ‘Text.show’,
              imported from ‘Data.Text’ at src/PlutusLedgerApi/Common/Eval.hs:45:1-24.
    |
117 |             ["Internal error: ", show ll, " does not support protocol version ", show pv]
    |                                  ^^^^

src/PlutusLedgerApi/Common/Eval.hs:117:82: error: [GHC-87543]
    Ambiguous occurrence ‘show’.
    It could refer to
       either ‘PlutusCore.show’,
              imported from ‘PlutusCore’ at src/PlutusLedgerApi/Common/Eval.hs:23:1-17
              (and originally defined in ‘GHC.Show’),
           or ‘Text.show’,
              imported from ‘Data.Text’ at src/PlutusLedgerApi/Common/Eval.hs:45:1-24.
    |
117 |             ["Internal error: ", show ll, " does not support protocol version ", show pv]
    |                                                                                  ^^^^
```

It seems that `Data.Text` added a `show` function in `2.1.2` and `PlutusLedgerApi.Common.Eval` has

```haskell
import Data.Text as Text
```

A similar problem occurs in `AnyProgram.Run`.

### Steps to reproduce the behavior

```diff
diff --git a/cabal.project b/cabal.project
index 22be30739..a388d351e 100644
--- a/cabal.project
+++ b/cabal.project
@@ -14,7 +14,7 @@ repository cardano-haskell-packages
 -- update either of these.
 index-state:
   -- Bump both the following dates if you need newer packages from Hackage
-  , hackage.haskell.org 2024-10-16T00:00:00Z
+  , hackage.haskell.org 2024-11-20T00:00:00Z
   -- Bump this if you need newer packages from CHaP
   , cardano-haskell-packages 2024-10-16T00:00:00Z
 
@@ -99,3 +99,4 @@ constraints:
   -- The API has changed for version 2.2, ledger depends on the old version and ledger will not
   -- be updated until after the Conway release.
   , cardano-crypto-class ^>= 2.1
+  , any.text source
diff --git a/flake.lock b/flake.lock
index 2b0526946..5c3df2eaa 100644
--- a/flake.lock
+++ b/flake.lock
@@ -279,11 +279,11 @@
     "hackage": {
       "flake": false,
       "locked": {
-        "lastModified": 1729124899,
-        "narHash": "sha256-cmb4iMcgk5+jUGMiZGNMzPCAnG17Kz9J6WIitYM17Fc=",
+        "lastModified": 1732148935,
+        "narHash": "sha256-zRj4ndWzyY9JFG3c8SCIxlFHefgS0YEazpq580gE3dk=",
         "owner": "input-output-hk",
         "repo": "hackage.nix",
-        "rev": "138edf81c8bcc4209e9706966f7feece70c37a96",
+        "rev": "834ac92115c32dabf23abd6b61f925e4d6edc9db",
         "type": "github"
       },
       "original": {
```

The `any.text source` constraint is copied from `cardano-api` where I first noticed the problem during release integration.


### Actual Result

See the error message in the description.

### Expected Result

Build succeeds.

### Describe the approach you would take to fix this

Using `import Data.Text as Text hiding (show)` isn't ideal because it prevents building with an older version of `text`. Ideally, `Data.Text` would be imported qualified, but I assume the team prefers it done unqualified. So the remaining option is to qualify the use of `show` as `PlutusCore.show`.

This problem doesn't need to be solved right away, because we can work around it in `cardano-api` with an additional constraint, `text <2.1.2`. However, it would be better in the long term if this was fixed, since having to add a constraint in a different project to make this one build isn't ideal.

### System info

OS: NixOS
Version: unstable
Plutus: `1.33.1.0-159-gefaadc5ce`

--- a/plutus-core/executables/plutus/AnyProgram/Run.hs
+++ b/plutus-core/executables/plutus/AnyProgram/Run.hs
@@ -9,6 +9,7 @@ import AnyProgram.IO
 import AnyProgram.With
 import Common
 import Control.Monad
+import Data.Text (unpack)
 import GetOpt
 import PlutusCore as PLC
 import PlutusCore.Evaluation.Machine.Ck as PLC
@@ -18,8 +19,6 @@ import PlutusPrelude
 import Types
 import UntypedPlutusCore as UPLC
 import UntypedPlutusCore.Evaluation.Machine.Cek as UPLC
-
-import Data.Text as Text
 
 runRun :: (?opts :: Opts)
        => SLang s -> FromLang s -> IO ()
@@ -48,10 +47,10 @@ runPlc (PLC.Program _ _ t)
       -- CK machine currently only works with ann==() , so we void before
       case PLC.runCk defaultBuiltinsRuntimeForTesting False (void t) of
         (Left errorWithCause, logs) -> do
-            for_ logs (printE . Text.unpack)
+            for_ logs (printE . unpack)
             failE $ show errorWithCause
         (Right finalTerm, logs) -> do
-            for_ logs (printE . Text.unpack)
+            for_ logs (printE . unpack)
             printE "Execution succeeded. Final Term:"
             -- TODO: lift the final term back to the target singleton
             printE "Execution succeeded. Final Term:"
@@ -64,10 +63,10 @@ runUplc :: (?opts :: Opts, Typeable a)
 runUplc (UPLC.UnrestrictedProgram (UPLC.Program _ _ t)) =
     case UPLC.runCekDeBruijn defaultCekParametersForTesting exBudgetMode logEmitter t of
         (Left errorWithCause, _, logs) -> do
-            for_ logs (printE . Text.unpack)
+            for_ logs (printE . unpack)
             failE $ show errorWithCause
         (Right finalTerm, finalBudget, logs) -> do
-            for_ logs (printE . Text.unpack)
+            for_ logs (printE . unpack)
             -- TODO: lift the final term back to the target singleton
             printE "Execution succeeded. Final Term:"
             printE $ show $ prettyWithStyle (_prettyStyle ?opts) finalTerm

--- a/plutus-ledger-api/src/PlutusLedgerApi/Common/Eval.hs
+++ b/plutus-ledger-api/src/PlutusLedgerApi/Common/Eval.hs
@@ -42,7 +42,7 @@ import Control.Monad.Error.Lens
 import Control.Monad.Except (MonadError (..), liftEither, runExceptT)
 import Control.Monad.Writer (MonadWriter (..), runWriter)
 import Data.Set as Set
-import Data.Text as Text
+import Data.Text (Text)
 import Data.Tuple
 import NoThunks.Class
 
@@ -80,7 +80,7 @@ data VerboseMode =
 
 It will be an empty list if the `VerboseMode` is set to `Quiet`.
 -}
-type LogOutput = [Text.Text]
+type LogOutput = [Text]
 
 {-| Shared helper for the evaluation functions: 'evaluateScriptCounting' and 'evaluateScriptRestricting',
 

